frontend/src/features/statistics/StatsContext.tsx

```diff
@@ -0,0 +1,91 @@
+import React, {
+  createContext,
+  useCallback,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { Statistics, StatsService } from 'src/services/openapi';
+
+// Stats are considered outdated after this amount of miliseconds.
+const EXPIRATION_TIME = 4000;
+
+interface StatsContextValue {
+  stats: Statistics;
+  getStats: () => Statistics;
+  refreshStats: () => void;
+}
+
+const initialState: Statistics = {
+  active_users: {
+    total: 0,
+    joined_last_month: 0,
+    joined_last_30_days: 0,
+  },
+  polls: [],
+};
+
+const StatsContext = createContext<StatsContextValue>({
+  stats: initialState,
+  getStats: () => initialState,
+  refreshStats: () => undefined,
+});
+
+export const StatsLazyProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const loading = useRef(false);
+  const lastRefreshAt = useRef(0);
+
+  const [stats, setStats] = useState(initialState);
+
+  const refreshStats = useCallback(async () => {
+    setStats(await StatsService.statsRetrieve());
+    loading.current = false;
+    lastRefreshAt.current = Date.now();
+  }, []);
```

Comment:
As `setStats()` will trigger a re-render, and possibly reevaluate `getStats()`, the references should be updated beforehand:

```suggestion
  const refreshStats = useCallback(async () => {
    const newStats = await StatsService.statsRetrieve();
    loading.current = false;
    lastRefreshAt.current = Date.now();
    setStats(newStats)
  }, []);
```